### ISSUE TITLE
Contrast issue with progress bars

### ISSUE THREAD

**Issue by HeyMyian** [vkbo/novelWriter#2884]
Talking about colors, the percentage numbers on the new progress bars are not always readable, especially around 50% for the daily goal (because `yellow` is always difficult in bright themes..) and the project goal on dark themes, because of lacking contrast between `blue` and the text color.

I have no good idea right now how to fix this that will also look appealing, without adding new color definitions for the bars. I figure being able to read the % number is not *that* important since the progress bar is the primary visual cue. But I also suppose when there is text, it ought to be readable. Maybe removing the text on the progress bar and putting it into the tooltip? Or putting the % number outside of the bar?

<img width="325" height="30" alt="Image" src="https://github.com/user-attachments/assets/49750505-539c-40b6-b276-fbbad6654926" />

(Screenshot from "Aura" dark theme)

**Comment by vkbo** [comment]
Yeah, this one is tricky. When the bar is not behind it, it uses the regular GUI text colour, but it overlays pure white on the bar when it runs under the text. I picked white because it matches all themes I tested, but on some it may be smarter with black on the bright themes. I just didn't like the pure black because it is too black for most themes.

I can try again by switching to using the base colour when the other colours are lighter than a certain threshold.

**Comment by vkbo** [comment]
I changed it so the base colour is always used on top of the coloured bar. That seems to work OK for all light and dark themes, but it's not 100% optimal on all.

**Comment by Survivian** [comment]
Question: Does QT not allow for text outlines or drop shadows? If possible, wouldn't the ideal solution to this be to style the percentage to always have the bar's background color as a text outline so that it's always at maximum readability?

**Comment by vkbo** [comment]
This progress bar is drawn manually in the code, so I would have to simulate the effect by drawing the text multiple times with an offset I think.

**Comment by Survivian** [comment]
Dang. If blurring text in real-time is an option, could always just make one more layer and then blur that text to make it at least a *little* more readable, but at this point, I'm just grasping at straws.
